File: bench/py/88_elem_float_compound.py
# Twin of bench/my/88_elem_float_compound.my. All values positive, so
# Python's floored float % agrees with MyLang's (and C's fmod).
import sys
scale = int(sys.argv[1]) if len(sys.argv) > 1 else 1

N = 400000 * scale
f = [i * 0.75 + 0.5 for i in range(64)]

for t in range(N):
    j = t & 63
    f[j] += 1.25 + (t & 7) * 0.375
    f[j] *= 0.5
    f[j] -= 0.125
    f[j] /= 1.5
    f[j] %= 1.0

s = 0.0
for v in f:
    s += v
print("result:", round(s, 4))

File: bench/cpp/88_elem_float_compound.cpp
/* SPDX-License-Identifier: BSD-2-Clause */
/* Faithful C++ of bench/my/88_elem_float_compound.my: compound float stores
 * on array elements. All values positive, so fmod agrees with Python's
 * floored % and MyLang's. The sum accumulates in source order. */
#include "bench.h"
#include <cmath>

int main(int argc, char **argv)
{
    long scale = bench_scale(argc, argv);
    long N = 400000L * scale;
    double f[64];

    for (long i = 0; i < 64; i++)
        f[i] = i * 0.75 + 0.5;

    for (long t = 0; t < N; t++) {
        long j = t & 63;
        f[j] += 1.25 + (double)(t & 7) * 0.375;
        f[j] *= 0.5;
        f[j] -= 0.125;
        f[j] /= 1.5;
        f[j] = fmod(f[j], 1.0);
    }

    double s = 0.0;
    for (long i = 0; i < 64; i++)
        s += f[i];
    printf("result: %.4f\n", s);
    return 0;
}
